luarules/gadgets.lua

```diff
@@ -812,17 +820,35 @@ function gadgetHandler:UpdateCallIn(name)
 		local selffunc = self[name]
 
 		if selffunc ~= nil then
-			_G[name] = function(...)
-				callinDepth = callinDepth + 1
-
-				local res = selffunc(self, ...)
+			local noUnpack = true
+			if table.contains(multiReturnTopCallins, name) then
+				noUnpack = false
+			end
 
+			local postOps = function()
 				callinDepth = callinDepth - 1
 				if reorderNeeded and callinDepth == 0 then
 					self:PerformReorders()
 				end
+			end
+
+			if noUnpack then
+				_G[name] = function(...)
+					callinDepth = callinDepth + 1
+
+					local res = selffunc(self, ...)
+					postOps()
+					return res
+				end
+			else
+				-- methods returning several values need them unpacked
+				_G[name] = function(...)
+					callinDepth = callinDepth + 1
 
-				return res
+					local res = {selffunc(self, ...)}
```

Comment:
Can a table and the unpack call be avoided? All of those callins return 2 values iirc so it could just be
```
local r1, r2 = selffunc(self, ...)
postOps()
return r1, r2
```
With the `unpack({foo})` reserved only for truly variadic callins

Reply:
Ok changed it to just 2 return values for now.

I'm a bit wary of using two retvals where only one is in use since not sure if behaviour might change somewhere, I guess not but on an initial test with 3 vars for everything something didn't seem right, but maybe I just messed up that test.